packages/editor/src/components/widgets/code-editor/useCodeEditor.ts

```diff
@@ -39,6 +39,38 @@ export const useMonacoEditor = (options?: { modifyAction?: ModifyAction; scriptA
     } else {
       const text = value.length > 0 && options?.modifyAction ? options.modifyAction(value) : value;
       editor.executeEdits('browser', [{ range: selection, text, forceMoveMarkers: true }]);
+      if (type === 'func') {
+        let lengthUntilOpeningParenthesis = 0;
+        for (let i = 0; i < text.length; i++) {
+          if (text[i] === '(') {
+            break;
+          }
+          lengthUntilOpeningParenthesis++;
+        }
+
+        const updatedEditorContent = editor.getValue();
+        const editorModel = editor.getModel();
+
+        if (editorModel && lengthUntilOpeningParenthesis !== text.length) {
+          const textIndex = updatedEditorContent.indexOf(text);
+          const startLineNumber = editorModel.getPositionAt(textIndex).lineNumber;
+          const startColumn = editorModel.getPositionAt(textIndex + 1 + lengthUntilOpeningParenthesis).column;
+          const endLineNumber = editorModel.getPositionAt(textIndex + text.length).lineNumber;
+          const endColumn = editorModel.getPositionAt(textIndex + text.length - 1).column;
+
+          const textrange = {
+            startLineNumber: startLineNumber,
+            startColumn: startColumn,
+            endLineNumber: endLineNumber,
+            endColumn: endColumn
+          };
+
+          setTimeout(() => {
+            editor.setSelection(textrange);
+            editor.focus();
+          }, 500);
```

Comment:
why is this timeout needed?

Reply:
I need the timeout because otherwise the focus is not set. Without the timeout the focus is on the button to open the browser after the "apply" and not on the monaco editor itself. It is a similar behavior as with the focushanldig of the search input, where I also had to work with a timeout to ensure that the corresponding command is not overwritten.

Reply:
I think we could definitly listen on ended transitions and only then set focus stuff, e.g. like it is described here: https://developer.mozilla.org/en-US/docs/Web/API/Element/transitionend_event 
or the same for animation events: https://developer.mozilla.org/en-US/docs/Web/API/Element/animationend_event